src/query/sql/src/planner/optimizer/decorrelate/decorrelate.rs

```diff
@@ -517,4 +523,221 @@ impl SubqueryRewriter {
+            if let RelOperator::EvalScalar(eval) = subquery.subquery.plan() {
+                if eval.items.len() != 1 {
+                    return Ok(None);
+                }
+                if let ScalarExpr::BoundColumnRef(bound_column) = &eval.items[0].scalar {
+                    output_column_index = Some(bound_column.column.index);
+                }
+            }
+            if output_column_index.is_none() {
+                return Ok(None);
+            }
+            let output_column_index = output_column_index.unwrap();
+
+            let mut srf_column_index = None;
+            if let RelOperator::EvalScalar(eval) = child.plan() {
+                if eval.items.len() != 1 || eval.items[0].index != output_column_index {
+                    return Ok(None);
+                }
+                if let ScalarExpr::FunctionCall(get_func) = &eval.items[0].scalar {
+                    if get_func.func_name == "get"
+                        && get_func.arguments.len() == 1
+                        && get_func.params.len() == 1
+                        && get_func.params[0] == Scalar::Number(NumberScalar::Int64(1))
+                    {
+                        if let ScalarExpr::BoundColumnRef(bound_column) = &get_func.arguments[0] {
+                            srf_column_index = Some(bound_column.column.index);
+                        }
+                    }
+                }
+            }
+            if srf_column_index.is_none() {
+                return Ok(None);
+            }
+            let srf_column_index = srf_column_index.unwrap();
+
+            let project_set_expr = child.child(0)?;
+            if let RelOperator::ProjectSet(project_set) = project_set_expr.plan() {
+                if project_set.srfs.len() != 1
+                    || project_set.srfs[0].index != srf_column_index
+                    || subquery.compare_op != Some(ComparisonOp::Equal)
+                    || subquery.typ != SubqueryType::Any
+                {
+                    return Ok(None);
+                }
+                let Ok(srf) = FunctionCall::try_from(project_set.srfs[0].scalar.clone()) else {
+                    return Ok(None);
+                };
+                if srf.arguments.len() != 1 {
+                    return Ok(None);
+                }
+                let Ok(const_scalar) = ConstantExpr::try_from(srf.arguments[0].clone()) else {
+                    return Ok(None);
+                };
+                let Some(child_expr) = &subquery.child_expr else {
+                    return Ok(None);
+                };
+                match &const_scalar.value {
+                    Scalar::EmptyArray => {
+                        return Ok(Some(ScalarExpr::ConstantExpr(ConstantExpr {
+                            span: subquery.span,
+                            value: Scalar::Null,
+                        })));
+                    }
+                    Scalar::Array(array_column) => {
```

Comment:
what if array_column has large length.

Need to respect `max_inlist_to_or` setting